test_tf.py

```diff
@@ -0,0 +1,153 @@
+import os
+import subprocess
+import unittest
+from main import get_policy_check_type, get_required_inputs, get_optional_inputs, get_sub_command
+from main import get_treat_findings_as_non_blocking_flag, build_command, execute_command, set_output
+
+CLI_POLICY_VALIDATOR = "tf-policy-validator"
+
+POLICY_CHECK_TYPE = "INPUT_POLICY-CHECK-TYPE"
+
+NOT_VALIDATE_POLICY = "NOT_VALIDATE_POLICY"
+VALIDATE_POLICY = "VALIDATE_POLICY"
+CHECK_NO_NEW_ACCESS = "CHECK_NO_NEW_ACCESS"
+CHECK_ACCESS_NOT_GRANTED = "CHECK_ACCESS_NOT_GRANTED"
+
+COMMON_REQUIRED_INPUTS = {"INPUT_TEMPLATE-PATH", "INPUT_REGION"}
+
+TREAT_FINDINGS_AS_NON_BLOCKING = "INPUT_TREAT-FINDINGS-AS-NON-BLOCKING"
+
+# assertRaises(SomeCoolException, mymod.myfunc)
+class TfpvTest(unittest.TestCase):
+    # test_get_type_case1: failure expected because required os.environ[] are not set")
+    @unittest.expectedFailure
+    def test_get_type_case1(self):
+        assertRaises(ValueError, get_policy_check_type)
+
+    # test_get_type_case2: failure expected because os.environ[] is set to an invalid value")
+    @unittest.expectedFailure
+    def test_get_type_case2(self):
+        os.environ[POLICY_CHECK_TYPE] = NOT_VALIDATE_POLICY
+        assertRaises(ValueError, get_policy_check_type)
+
+    def test_get_type_case3(self):
+        os.environ[POLICY_CHECK_TYPE] = VALIDATE_POLICY
+        policy_type = get_policy_check_type()
+        assert policy_type == VALIDATE_POLICY
+
+    # test_get_required_input_case4: failure expected because an invalid policy_check_type is provided")
+    @unittest.expectedFailure
+    def test_get_required_input_case4(self):
+        policy_check = "INVALIDATE_POLICY"
+        result = get_required_inputs(policy_check)
+        assert result == {"INPUT_TEMPLATE-PATH",  "INPUT_REGION"}
```

Comment:
might be better to assert None instead of letting the expect failure to capture it.

Reply:
will consider.